File: src/main/java/com/flexpoker/table/query/repository/impl/InMemoryCardsUsedInHandRepository.java
package com.flexpoker.table.query.repository.impl;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

import org.springframework.stereotype.Repository;

import com.flexpoker.model.card.FlopCards;
import com.flexpoker.model.card.PocketCards;
import com.flexpoker.model.card.RiverCard;
import com.flexpoker.model.card.TurnCard;
import com.flexpoker.table.query.repository.CardsUsedInHandRepository;

@Repository
public class InMemoryCardsUsedInHandRepository implements CardsUsedInHandRepository {

    private final Map<UUID, FlopCards> handIdToFlopCardsMap;

    private final Map<UUID, TurnCard> handIdToTurnCardMap;

    private final Map<UUID, RiverCard> handIdToRiverCardMap;

    private final Map<UUID, Map<UUID, PocketCards>> handIdToPocketCardsMap;

    public InMemoryCardsUsedInHandRepository() {
        handIdToFlopCardsMap = new HashMap<>();
        handIdToTurnCardMap = new HashMap<>();
        handIdToRiverCardMap = new HashMap<>();
        handIdToPocketCardsMap = new HashMap<>();
    }

    @Override
    public void saveFlopCards(UUID handId, FlopCards flopCards) {
        handIdToFlopCardsMap.put(handId, flopCards);
    }

    @Override
    public void saveTurnCard(UUID handId, TurnCard turnCard) {
        handIdToTurnCardMap.put(handId, turnCard);
    }

    @Override
    public void saveRiverCard(UUID handId, RiverCard riverCard) {
        handIdToRiverCardMap.put(handId, riverCard);
    }

    @Override
    public void savePocketCards(UUID handId, Map<UUID, PocketCards> playerToPocketCardsMap) {
        handIdToPocketCardsMap.put(handId, playerToPocketCardsMap);
    }

    @Override
    public FlopCards fetchFlopCards(UUID handId) {
        return handIdToFlopCardsMap.get(handId);
    }

    @Override
    public TurnCard fetchTurnCard(UUID handId) {
        return handIdToTurnCardMap.get(handId);
    }

    @Override
    public RiverCard fetchRiverCard(UUID handId) {
        return handIdToRiverCardMap.get(handId);
    }

    @Override
    public PocketCards fetchPocketCards(UUID handId, UUID playerId) {
        return handIdToPocketCardsMap.get(handId).get(playerId);
    }

}

File: src/main/kotlin/com/flexpoker/table/query/repository/impl/InMemoryCardsUsedInHandRepository.kt
package com.flexpoker.table.query.repository.impl

import com.flexpoker.config.ProfileNames
import com.flexpoker.table.command.FlopCards
import com.flexpoker.table.command.PocketCards
import com.flexpoker.table.command.RiverCard
import com.flexpoker.table.command.TurnCard
import com.flexpoker.table.query.repository.CardsUsedInHandRepository
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Repository
import java.util.UUID

@Profile(ProfileNames.DEFAULT, ProfileNames.TABLE_QUERY_INMEMORY)
@Repository
class InMemoryCardsUsedInHandRepository : CardsUsedInHandRepository {

    private val handIdToFlopCardsMap: MutableMap<UUID, FlopCards> = HashMap()
    private val handIdToTurnCardMap: MutableMap<UUID, TurnCard> = HashMap()
    private val handIdToRiverCardMap: MutableMap<UUID, RiverCard> = HashMap()
    private val handIdToPocketCardsMap: MutableMap<UUID, Map<UUID, PocketCards>> = HashMap()

    override fun saveFlopCards(handId: UUID, flopCards: FlopCards) {
        handIdToFlopCardsMap[handId] = flopCards
    }

    override fun saveTurnCard(handId: UUID, turnCard: TurnCard) {
        handIdToTurnCardMap[handId] = turnCard
    }

    override fun saveRiverCard(handId: UUID, riverCard: RiverCard) {
        handIdToRiverCardMap[handId] = riverCard
    }

    override fun savePocketCards(handId: UUID, playerToPocketCardsMap: Map<UUID, PocketCards>) {
        handIdToPocketCardsMap[handId] = playerToPocketCardsMap
    }

    override fun fetchFlopCards(handId: UUID): FlopCards {
        return handIdToFlopCardsMap[handId]!!
    }

    override fun fetchTurnCard(handId: UUID): TurnCard {
        return handIdToTurnCardMap[handId]!!
    }

    override fun fetchRiverCard(handId: UUID): RiverCard {
        return handIdToRiverCardMap[handId]!!
    }

    override fun fetchPocketCards(handId: UUID, playerId: UUID): PocketCards {
        return handIdToPocketCardsMap[handId]!![playerId]!!
    }

    override fun fetchAllPocketCardsForUser(playerId: UUID): Map<UUID, PocketCards> {
        return handIdToPocketCardsMap
            .filter { it.value.containsKey(playerId) }
            .mapValues { it.value[playerId]!! }
    }

}